src/main.rs

```diff
@@ -120,7 +125,40 @@ fn main() {
 
     let build_server = remote.host;
 
-    let build_path = project_dir.clone();
+    let build_path = if args.use_temp_dir {
+        // Create a temporary directory on the remote server using mktemp
+        let project_name = project_dir.file_name().unwrap_or("unnamed-project");
```

Comment:
Should this `file_name` call ever realistically return `Err`? If not, maybe we should log an error (or even panic) instead of silently falling back.